client/register.go

```diff
@@ -43,11 +44,28 @@ var registerTimeout = cmdRegister.Flag.Int("t", 5, "")
 const registerRecheckTime = 10 * time.Millisecond
 
 func runRegister(cmd *Command, args []string) {
-	if *registerKey == "" && *registerKeyFile == "" {
+	k := NewKeysFile(path.Join(daemonFolder, daemonToRegister))
+	if *registerRemove && *registerKey == "" && *registerKeyFile == "" {
+		// Short circuit & handle `knox register -r`, which is expected to remove all keys
+		err := k.Lock()
+		if err != nil {
+			fatalf("There was an error obtaining file lock: %s", err.Error())
+		}
+		err = k.Overwrite([]string{})
+		if err != nil {
```

Comment:
log the error (from Overwrite) before trying to Unlock()? Error may be missed if Unlock() also produces error.